R/git.R

```diff
@@ -323,85 +323,117 @@ git_clean <- function() {
+  if ("git" %in% tool && "global" %in% scope) {
+    cli::cli_h3("Git (global)")
+    kv_line("Name", git_cfg_get("user.name", "global"))
+    kv_line("Email", git_cfg_get("user.email", "global"))
+    kv_line("Global (user-level) gitignore file", git_ignore_path("user"))
+    vaccinated <- git_vaccinated()
+    kv_line("Vaccinated", vaccinated)
+    if (!vaccinated) {
+      ui_info("See {ui_code('?git_vaccinate')} to learn more")
+    }
+    kv_line("Default Git protocol", git_protocol())
+    kv_line("Default initial branch name", init_default_branch)
+  }
 
   # github (global / user) -----------------------------------------------------
-  hd_line("GitHub")
   default_gh_host <- get_hosturl(default_api_url())
-  kv_line("Default GitHub host", default_gh_host)
-  pat_sitrep(default_gh_host)
+  if ("github" %in% tool && "global" %in% scope) {
+    cli::cli_h3("GitHub (global)")
+    kv_line("Default GitHub host", default_gh_host)
+    pat_sitrep(default_gh_host)
+  }
 
   # git and github for active project ------------------------------------------
-  hd_line("Git repo for current project")
+  if ("project" %in% scope) {
 
-  if (!proj_active()) {
-    ui_info("No active usethis project")
-    return(invisible())
-  }
-  kv_line("Active usethis project", proj_get())
-  if (!uses_git()) {
-    ui_info("Active project is not a Git repo")
-    return(invisible())
-  }
+    if (!proj_active()) {
+      ui_info("No active usethis project")
+      return(invisible())
+    }
+    kv_line("Active usethis project", proj_get())
+    if (!uses_git()) {
+      ui_info("Active project is not a Git repo")
+      return(invisible())
+    }
+
+    # current branch -------------------------------------------------------------
+    branch <- tryCatch(git_branch(), error = function(e) NULL)
+    tracking_branch <- if (is.null(branch)) NA_character_ else git_branch_tracking()
+    # TODO: can't really express with kv_line() helper
+    branch <- if (is.null(branch)) "<unset>" else branch
+    tracking_branch <- if (is.na(tracking_branch)) "<unset>" else tracking_branch
+
+    # local git config -----------------------------------------------------------
+    if ("git" %in% tool) {
+      cli::cli_h3("Git (project)")
+      if (proj_active() && uses_git()) {
+        local_user <- list(
+          user.name = git_cfg_get("user.name", "local"),
+          user.email = git_cfg_get("user.email", "local")
+        )
+        if (!is.null(local_user$user.name) || !is.null(local_user$user.email)) {
+          ui_info("This repo has a locally configured user")
+        }
+        if (!is.null(local_user$user.name)) {
+          kv_line("Name", local_user$user.name)
+        }
+        if (!is.null(local_user$user.email)) {
+          kv_line("Email", local_user$user.email)
+        }
+      }
+
+      # default branch -------------------------------------------------------------
+      default_branch_sitrep()
```

Comment:
This is now in a gray area, because it straddles the boundary between git and GitHub config for the local repo.

Maybe we should check if the active project uses GitHub.

If not: report default branch, (if `"github" %in% tool`) report GitHub remote config (which will be `"no_github"`, by definition). And (I think?) we're done. Exit.

If yes: use the existing code.

Reply:
I took a swing at this in 08344f95, hoping I've got it "right". Happy to take another swing if not.

Reply:
Here's what a git-but-no-github response looks like:

```
── Active usethis project: '/Users/ijlyttle/repos/public/forked/temp' ─────────────────────────────────────────────────────────

── Git local (project) 
✖ Default branch cannot be determined.
• Current local branch -> remote tracking branch:
  '<unset>' -> '<unset>'

── GitHub project 
ℹ Project does not use GitHub
```

Reply:
I'm sort of surprised that it says default branch cannot be determined. Is this a brand new repo that doesn't even have an initial commit? I think that might explain it.